source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -26,40 +26,66 @@ FileImplWin32::~FileImplWin32() {
   }
 }
 
-void FileImplWin32::openFile(FlagSet in) {
-  const auto flags_and_mode = translateFlag(in);
-  fd_ = ::open(path_.c_str(), flags_and_mode.flags_, flags_and_mode.pmode_);
+Api::IoCallBoolResult FileImplWin32::open(FlagSet in) {
+  if (isOpen()) {
+    return resultSuccess(true);
+  }
+
+  auto flags = translateFlag(in);
+  fd_ = CreateFileA(path_.c_str(), flags.access_, NULL, NULL, flags.creation_, NULL, NULL);
```

Comment:
Lets try to use 0 instead of NULL for the DWORD parameters

Reply:
Sure, will update. Out of curiosity what is the reason behind that?

Reply:
People usually expect NULL for pointers and not integers. If this repo has a style rule against that then by all means follow the repo style

Reply:
Gotcha. Thanks for the explanation.

Reply:
Fixed in e22ac3eaa32da79d7ce14dc1a0045ceb28fd4891